state-transition/core/state/statedb.go

```diff
@@ -96,17 +102,27 @@ func (s *StateDB) ExpectedWithdrawals(timestamp math.U64) (engineprimitives.With
 
 	withdrawalIndex, err := s.GetNextWithdrawalIndex()
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
 	validatorIndex, err := s.GetNextWithdrawalValidatorIndex()
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
 	totalValidators, err := s.GetTotalValidators()
 	if err != nil {
-		return nil, err
+		return nil, 0, err
+	}
+
+	// [New in Electra:EIP7251] Consume pending partial withdrawals
+	timestampFork := s.cs.ActiveForkVersionForTimestamp(timestamp)
+	if version.EqualsOrIsAfter(timestampFork, version.Electra()) {
```

Comment:
```suggestion
	forkVersion := s.cs.ActiveForkVersionForTimestamp(timestamp)
	if version.EqualsOrIsAfter(forkVersion, version.Electra()) {
```
just find timestampFork to be confusing. `forkVersion` is implied to be based on timestamp